File: lib/models/corrective_action_entry.dart
/// Entrada del historial de acciones correctivas sobre un registro.
class CorrectiveActionEntry {
  const CorrectiveActionEntry({
    required this.fecha,
    required this.responsable,
    required this.accion,
  });

  final DateTime fecha;
  final String responsable;
  final String accion;

  Map<String, dynamic> toJson() => {
        'fecha': fecha.toIso8601String(),
        'responsable': responsable,
        'accion': accion,
      };

  factory CorrectiveActionEntry.fromJson(Map<String, dynamic> json) {
    return CorrectiveActionEntry(
      fecha: DateTime.tryParse(json['fecha']?.toString() ?? '') ??
          DateTime.now(),
      responsable: json['responsable']?.toString() ?? '',
      accion: json['accion']?.toString() ?? '',
    );
  }
}

File: RegNeps.Net/src/RegNeps.Domain/Entities/CorrectiveActionEntry.cs
namespace RegNeps.Domain.Entities;

/// <summary>Entrada del historial de acciones correctivas.</summary>
public sealed class CorrectiveActionEntry
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid NepRecordId { get; set; }
    public NepRecord? NepRecord { get; set; }
    public string Accion { get; set; } = string.Empty;
    public string Responsable { get; set; } = string.Empty;
    public DateTime Fecha { get; set; } = DateTime.UtcNow;
}
